sdks/java/core/src/main/java/org/apache/beam/sdk/io/TFRecordIO.java

```diff
@@ -134,18 +133,22 @@ public Read withoutValidation() {
       return toBuilder().setValidate(false).build();
     }
 
+    /** @deprecated Use {@link #withCompression}. */
+    @Deprecated
+    public Read withCompressionType(TFRecordIO.CompressionType compressionType) {
+      return withCompression(compressionType.canonical);
+    }
+
     /**
-     * Returns a transform for reading TFRecord files that decompresses all input files
-     * using the specified compression type.
+     * Returns a transform for reading TFRecord files that decompresses all input files using the
+     * specified compression type.
      *
-     * <p>If no compression type is specified, the default is
-     * {@link TFRecordIO.CompressionType#AUTO}.
-     * In this mode, the compression type of the file is determined by its extension
-     * (e.g., {@code *.gz} is gzipped, {@code *.zlib} is zlib compressed, and all other
-     * extensions are uncompressed).
+     * <p>If no compression type is specified, the default is {@link Compression#AUTO}. In this
+     * mode, the compression type of the file is determined by its extension (e.g., {@code *.gz} is
+     * gzipped, {@code *.zlib} is zlib compressed, and all other extensions are uncompressed).
```

Comment:
<!--new_thread; commit:12876d43a961dd8ff9dba5f4cffcfbe643927eee; resolved:0-->
The comment seems like gz and zlib is all we support for auto detection. It would be better if it was reworded to say something more generic like:
`the compression type of the file is determined by its extension (e.g. {@code *.gz} is gzipped), if the extension is unknown it is considered to be uncompressed.`

Reply:
Done.